### PR TITLE
Add support for default transform names

### LINKED ISSUE #2760: Allow overriding transform default names
See https://github.com/spotify/scio/pull/2735#pullrequestreview-370466469 for context.

### PR DESCRIPTION
**Pull request by anish749**
Fixes #2760

### CODE REVIEW

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala [hunk 1]
```diff
@@ -279,7 +279,7 @@ final case class BigQueryTypedTable[T: Coder](
 
   override protected def read(sc: ScioContext, params: ReadP): SCollection[T] = {
     val io = reader.from(table.ref).withCoder(CoderMaterializer.beam(sc, Coder[T]))
-    sc.wrap(sc.pipeline.apply(s"Read BQ table ${table.spec}", io))
+    sc.wrap(sc.applyInternal(s"Read BQ table ${table.spec}", io))
```
Comment (regadas) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala]: I think eventually we will need to track down all the custom names we have and add this.

**Comment by regadas** [comment]
@anish749 Thanks for this!

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala (hunk 1 above)
Comment (anish749) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala]: I didn't find any other usage of this. This was the only place.

Comment (regadas) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala]: oh good to know! The good thing about this change is that we can now safely add more default human readable names to transforms! which is a big plus

**Pull request merged**